### PR TITLE
updated latest button with promise

### PR DESCRIPTION
**Pull request by jonfu86**
Please review before merging!

### CODE REVIEW

**Comment by thibaultzanini** [comment]
The branch has conflicts can you fix it?

File: ui/button.reel/button.js [hunk 1]
```diff
@@ -462,4 +495,4 @@ Button.addAttributes( /** @lends module:"montage/ui/native/button.reel".Button#
 */
     value: null
 
-});
+});
```
Comment (thibaultzanini) [inline ui/button.reel/button.js]: missing end line

**Comment by marchant** [comment]
It's missing tests, examples in .info, what happens if multiple 2 promises are set one after the other and the first one resolves?

**Comment by jonfu86** [comment]
updated with promise test and logic to handle multiple promises

File: ui/button.reel/button.js [hunk 2]
```diff
@@ -4,7 +4,8 @@
     @module "montage/ui/native/button.reel"
 */
 var Control = require("ui/control").Control,
-    PressComposer = require("composer/press-composer").PressComposer;
+    PressComposer = require("composer/press-composer").PressComposer,
+    Dict = require("collections/dict");
```
Comment (marchant) [inline ui/button.reel/button.js]: Dict doesn't seems to be used, and generally shouldn't be going forward

File: ui/button.reel/button.js [hunk 3]
```diff
@@ -83,20 +83,46 @@ var Button = exports.Button = Control.specialize(/** @lends module:"montage/ui/n
...
+        }
+    },
+
     /**
-        converter
         A Montage converter object used to convert or format the label displayed by the Button instance. When a new value is assigned to <code>label</code>, the converter object's <code>convert()</code> method is invoked, passing it the newly assigned label value.
         @type {Property}
         @default null
     */
+    converter: {
```
Comment (marchant) [inline ui/button.reel/button.js]: This already exists on Control, please remove the comments as it looks like it got left behind

File: ui/button.reel/button.js [hunk 4]
```diff
@@ -133,9 +158,8 @@ var Button = exports.Button = Control.specialize(/** @lends module:"montage/ui/n
                 }
             }
 
-            this._label = String(value);
-
-            if (this.isInputElement) {
+            this._label = "" + value;
```
Comment (marchant) [inline ui/button.reel/button.js]: please keep what was there: this._label = String(value); it avoids creating a new string if value is already one

**Pull request closed without merging**